src/clj/game/cards/ice.clj

```diff
@@ -1583,6 +1583,70 @@
+                                  (show-wait-prompt state :runner "Corp to decide on Sadaka R&D card actions")
+                                  (continue-ability
+                                    state side
+                                    {:prompt (str "Top 3 cards of R&D: " (clojure.string/join ", " c-names))
+                                     :choices ["Arrange cards" "Shuffle R&D"]
+                                     :effect
+                                     (req
+                                       (if (= target "Arrange cards")
+                                         (do
+                                           (when-completed
+                                             (resolve-ability state side (reorder-choice :corp c) card nil)
+                                             (do
+                                               (system-msg state :corp (str "rearranges the top " (count c) " cards of R&D"))
+                                               (clear-wait-prompt state :runner)
+                                               (maybe-draw state side card))))
+                                         (do
+                                           (shuffle! state :corp :deck)
+                                           (system-msg state :corp (str "shuffles R&D"))
+                                           (clear-wait-prompt state :runner)
+                                           (maybe-draw state side card))))}
+                                    card nil)))}
+                  {:label "You may trash 1 card in HQ. If you do, trash 1 resource. Trash Sadaka."
+                   :delayed-completion true
+                   :effect
+                   (req (show-wait-prompt state :runner "Corp to select cards to trash with Sadaka")
+                        (when-completed
+                          (resolve-ability
+                            state side
+                            {:prompt "Choose a card in HQ to trash"
+                             :choices (req (cancellable (:hand corp) :sorted))
+                             :cancel-effect (effect (system-msg "chooses not to trash a card from HQ"))
+                             :delayed-completion true
+                             :effect (req (trash state :corp target)
+                                          (system-msg state :corp "trashes a card from HQ")
+                                          (when-completed
+                                            (resolve-ability state side trash-resource-sub card nil)
+                                            (system-msg state :corp "done trashing cards")))}
+                            card nil)
+                          (do
+                            (trash state :corp card)
+                            (system-msg state :corp "trashes Sadaka")
+                            (clear-wait-prompt state :runner)
+                            (effect-completed state side eid))))}]}
```

Comment:
Move the `trash` after the msg and clear-wait, then pass your `eid` to it; you won't need effect-completed.